### PR TITLE
Check advanced networking loopback test compat

### PR DESCRIPTION
**Pull request by agirault**
- dynamically check for NIC address
- warn if not available and skip test

### CODE REVIEW

**Review by jjomier — APPROVED** [review]

File: applications/adv_networking_bench/cpp/CMakeLists.txt [hunk 1]
```diff
@@ -104,43 +104,54 @@ install(
...
+      message(STATUS "[ANO Test] Skipping advanced networking loopback test")
+      return()
+    else()
+      message(STATUS "[ANO Test] ${MSG} ${${OUTPUT_VAR}}")
+    endif()
+  endmacro()
+
+  # Check NIC addresses
+  message(STATUS "[ANO Test] Checking system compatibility for advanced networking loopback test")
+  execute_and_check("ConnectX BUS Addresses:" "lspci | grep -E \"Ethernet.*ConnectX\" | awk '{print $1}'" cx_bus_address_list)
```
Comment (agirault) [inline applications/adv_networking_bench/cpp/CMakeLists.txt:130]: Per @cliffburdick , will update to

```suggestion
  execute_and_check("NVIDIA NIC BUS Addresses:" "lspci -nn | grep -iE 'Ethernet controller.*\[15b3:' | awk '{print $1}'" cx_bus_address_list)
```

File: applications/adv_networking_bench/cpp/CMakeLists.txt [hunk 2]
```diff
@@ -104,43 +104,55 @@ install(
 
 # Add testing
 if(BUILD_TESTING)
-
-  ## adv_networking_bench_default_tx_rx_test ##
-  # The test below is expected to run on an IGX devkit with a loopback
-  # connecting both ports of the IGX NIC (default config currently hardcodes
-  # the BUS address of these two ports), with optimal system setups (BAR1,
+  ## adv_networking_bench_dpdk_loopback_test ##
```
Comment (cliffburdick) [inline applications/adv_networking_bench/cpp/CMakeLists.txt:107]: On the same topic as before, "loopback" is an overloaded term and in the future we may want multiple versions. IE true loopback with a cable (this config), hardware loopback without a cable (TBD), and software loopback (the other PR)

File: applications/adv_networking_bench/cpp/CMakeLists.txt [hunk 3]
```diff
@@ -104,43 +104,55 @@ install(
...
+      message(STATUS "[ANO Test] Skipping advanced networking loopback test")
+      return()
+    else()
+      message(STATUS "[ANO Test] ${MSG} ${${OUTPUT_VAR}}")
+    endif()
+  endmacro()
+
+  # Check NIC addresses
+  message(STATUS "[ANO Test] Checking system compatibility for advanced networking loopback test")
+  execute_and_check("NVIDIA NIC BUS Addresses:" "lspci -nn | grep -iE 'Ethernet controller.*15b3:' | awk '{print $1}'" cx_bus_address_list)
```
Comment (cliffburdick) [inline applications/adv_networking_bench/cpp/CMakeLists.txt:131]: How is "Ethernet controller" guaranteed to give only NVIDIA NICs?

File: applications/adv_networking_bench/cpp/CMakeLists.txt [hunk 4]
```diff
@@ -104,43 +104,55 @@ install(
...
+  string(REGEX REPLACE "address: [^\n]*Source[^\n]*Tx[^\n]*" "address: ${tx_bus}" TEST_CONFIG ${TEST_CONFIG})
+  string(REGEX REPLACE "eth_dst_addr:[^\n]*" "eth_dst_addr: ${rx_eth_mac}" TEST_CONFIG ${TEST_CONFIG})
+  file(WRITE "${CMAKE_CURRENT_BINARY_DIR}/adv_networking_bench_dpdk_loopback_test.yaml" ${TEST_CONFIG})
 
-  # Create test
+  # Define the test
   file(COPY "testing/adv_networking_test_harness.py" DESTINATION "testing")
   add_test(NAME adv_networking_bench_dpdk_test
-           COMMAND python3 testing/adv_networking_test_harness.py "./adv_networking_bench adv_networking_bench_default_tx_rx_test.yaml" --avg-throughput-threshold 90 --missed-packets-threshold 0.1
+           COMMAND python3 testing/adv_networking_test_harness.py "./adv_networking_bench adv_networking_bench_dpdk_loopback_test.yaml" --avg-throughput-threshold 90 --missed-packets-threshold 0.1
```
Comment (cliffburdick) [inline applications/adv_networking_bench/cpp/CMakeLists.txt:156]: I would change the file name to "multi_if_loopback" to indicate we're using multiple interfaces with a cable.

File: applications/adv_networking_bench/cpp/CMakeLists.txt (hunk 3 above)
Comment (agirault) [reply applications/adv_networking_bench/cpp/CMakeLists.txt:131]: - `Ethernet controller` is a PCI-SIG standard class (code `0200`) for all NICs
- `15b3` is Mellanox's vendor ID.

Comment (agirault) [reply applications/adv_networking_bench/cpp/CMakeLists.txt:131]: I'll switch to using `lspci -n` which provides the codes only, better for parsing.

Comment (agirault) [reply applications/adv_networking_bench/cpp/CMakeLists.txt:131]:
```
lspci -n | awk '$2 == "0200:" && $3 ~ /^15b3:/ {print $1}'
```

File: applications/adv_networking_bench/cpp/CMakeLists.txt (hunk 4 above)
Comment (agirault) [reply applications/adv_networking_bench/cpp/CMakeLists.txt]: ty, done

File: applications/adv_networking_bench/cpp/CMakeLists.txt (hunk 3 above)
Comment (agirault) [reply applications/adv_networking_bench/cpp/CMakeLists.txt]: https://github.com/nvidia-holoscan/holohub/compare/68ace7a713d73eb10a21f1d1eceaad63a2803244..f03f9e9496f4dd5bbdf87c5e395e6607e094d46e

File: applications/adv_networking_bench/cpp/CMakeLists.txt (hunk 2 above)
Comment (agirault) [reply applications/adv_networking_bench/cpp/CMakeLists.txt]: Renamed: https://github.com/nvidia-holoscan/holohub/compare/f03f9e9496f4dd5bbdf87c5e395e6607e094d46e..d94f65840789c265a83fe5b22768eb2cc7890a3d

**Review by cliffburdick — APPROVED** [review]

**Pull request merged**